### PR TITLE
Manufacture Delivery might be eligible.

### LINKED ISSUE #804: Delivery of marijuana to a minor is eligible
The bug report has the following as a type-eligible charge:

4758604A-Delivery of Marijuana to Minor

The eligibility is given under a different statute from the usual expungement statute:

https://www.oregonlaws.org/ors/137.226

It's probably best to create a new charge type for this, and it's possible that the charge type could include other marijuana charges, depending on the details in 137.226

### PR DESCRIPTION
**Pull request by wittejm**
Manu/del is identifiable with "Manu/del" or "delivery" keywords in the charge name.

This is only to pick of the possibility of having a marijuana charge that is not directly identifiable by name or statute.
 - Inspect manually to see if the charge meets eligibility as a marijuana charge
 - Allow fall-through to charge level charge types if it's schedule 2. 

Addresses #809 
closes #804 

Still needs fixed unit tests.

### CODE REVIEW

**Comment by wittejm** [comment]
Unit tests failing because leap day is coming up. That is seriously excellent.

**Comment by wittejm** [comment]
Adding also the MarijuanaEligible type, which could have a better name since it's similar to MarijuanaIneligible.

This adds some more exacting checks in the `charge_classifier` for drug/marijuana charges.

@michaelzhang43 said we can remove 475992 from PCS schedule 1 statutes , so that is breaking a few of our tests that I have not updated. We need to verify that we can handle eligibility correctly without relying on the statute number (it turns out to be very unindicative for eligibility).

Still a WIP because we should add more tests for the various possible records, particularly some real records that we can pull from the Bug Reports page or obtain from Michael.

**Comment by KentShikama** [comment]
It looks like some tests in https://github.com/codeforpdx/recordexpungPDX/blob/master/src/backend/tests/models/charge_types/test_schedule_1_p_c_s.py need to be adjusted.

**Comment by wittejm** [comment]
@KentShikama I think we need to have this change in for drug charges and it gets us 90% of the way, and the last 10% can be handled by quantum analysis. But please lmk if you have suggestions on this portion.

File: src/backend/tests/models/charge_types/test_manufacture_delivery.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    manufacture_delivery_charge = ChargeFactory.create(**charge_dict)
+
+    assert isinstance(manufacture_delivery_charge, ManufactureDelivery)
+    assert manufacture_delivery_charge.expungement_result.type_eligibility.status is EligibilityStatus.NEEDS_MORE_ANALYSIS
+    assert (
+        manufacture_delivery_charge.expungement_result.type_eligibility.reason == "This may be eligible if it is a charge for marijuana, under 137.226. See additional legal details."
+    )
+
+def test_manufacture_delivery_manudel_felony_unclassified():
+    charge_dict = ChargeFactory.default_dict()
```
Comment (KentShikama) [inline src/backend/tests/models/charge_types/test_manufacture_delivery.py]: This function no longer exists

File: src/backend/expungeservice/models/charge_types/marijuana_eligible.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+from dataclasses import dataclass
+
+from expungeservice.models.charge import Charge
+from expungeservice.models.expungement_result import TypeEligibility, EligibilityStatus
+from expungeservice.models.disposition import DispositionStatus
+
+
+@dataclass(eq=False)
```
Comment (KentShikama) [inline src/backend/expungeservice/models/charge_types/marijuana_eligible.py]:
```suggestion
@dataclass
```

File: src/backend/expungeservice/models/charge_types/manufacture_delivery.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+from dataclasses import dataclass
+
+from expungeservice.models.charge import Charge
+from expungeservice.models.expungement_result import TypeEligibility, EligibilityStatus
+
+
+@dataclass(eq=False)
```
Comment (KentShikama) [inline src/backend/expungeservice/models/charge_types/manufacture_delivery.py]:
```suggestion
@dataclass
```

**Pull request merged**